pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1336,22 +1340,70 @@ public void openLedgerFailed(ManagedLedgerException exception, Object ctx) {
+                        Optional<GetResult> racksData =
+                                this.pulsar.getLocalMetadataStore().
+                                        get(ZkBookieRackAffinityMapping.BOOKIE_INFO_ROOT_PATH).join();
+                        if (racksData.isPresent()) {
+                            try {
+                                BookiesRackConfiguration allGroupsBookieMapping =
+                                        ObjectMapperFactory.getThreadLocal().
+                                                readValue(racksData.get().getValue(), BookiesRackConfiguration.class);
+                                if (allGroupsBookieMapping.size() > 0) {
+                                    managedLedgerConfig
+                                            .setBookKeeperEnsemblePlacementPolicyClassName(
+                                                    ZkIsolatedBookieEnsemblePlacementPolicy.class);
+                                    if (localPolicies.isPresent()
+                                            && localPolicies.get().bookieAffinityGroup != null) {
+                                        Map<String, Object> properties = Maps.newHashMap();
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                        .ISOLATION_BOOKIE_GROUPS,
+                                                localPolicies.get().bookieAffinityGroup
+                                                        .getBookkeeperAffinityGroupPrimary());
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                        .SECONDARY_ISOLATION_BOOKIE_GROUPS,
+                                                localPolicies.get().bookieAffinityGroup
+                                                        .getBookkeeperAffinityGroupSecondary());
+                                        managedLedgerConfig.
+                                                setBookKeeperEnsemblePlacementPolicyProperties(properties);
+                                    } else {
+                                        Map<String, Object> properties = Maps.newHashMap();
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                .ISOLATION_BOOKIE_GROUPS, "");
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                .SECONDARY_ISOLATION_BOOKIE_GROUPS, "");
+                                        managedLedgerConfig.
+                                                setBookKeeperEnsemblePlacementPolicyProperties(properties);
+                                    }
+                                }
+                            } catch (Exception e) {
+                                log.warn("Error getting bookie isolation info from zk", e);
```

Comment:
we should fail,
otherwise the broker will continue with a wrong configuration
and we will push the data to the wrong bookies probably